server/src/tree-sitter/analyzer.ts

```diff
@@ -71,12 +71,11 @@ export default class Analyzer {
     return this.uriToAnalyzedDocument[uri]
   }
 
+  /**
+  * Get the scan result for the current file. If a recipe provided, it will return the scan result of that recipe. Otherwise, it will return the global env scan result.
```

Comment:
I see, then together with the comment you put above on the command. I think what we want is:
Show `global scan` option and hide `recipe scan` option when it is a `.conf` or `.bbclass`.
Show `recipe scan` option and hide `global scan` option when it is a `.bb`, `.inc` or `.bbappend`.

Is that correct?

Reply:
Yes this would work to, but I'm not sure you can make that distinction in the package.json. All extensions are registered under the same type in our package.json

Reply:
It is feasible. Check out the new commit.